airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py

```diff
@@ -164,3 +169,37 @@ def _iter(self, x: Iterable[Any]) -> Iterator[Any]:
             if isinstance(item, Exception):
                 raise item
             yield item
+
+
+class TestFileBasedErrorCollector:
+
+    test_error_collector: FileBasedErrorsCollector = FileBasedErrorsCollector()
+
+    @pytest.mark.parametrize(
+        "stream, file, line_no, n_skipped",
+        (
+            ("stream_1", "test.csv", 1, 1),
+        ),
+    )
+    def test_collect_parsing_error(self, stream, file, line_no, n_skipped) -> None:
+        test_error_pattern = "Error parsing record."
+        # format the error body
+        test_error = AirbyteMessage(
+            type=MessageType.LOG,
+            log=AirbyteLogMessage(
+                level=Level.ERROR,
+                message=f"{FileBasedSourceError.ERROR_PARSING_RECORD.value} stream={stream} file={file} line_no={line_no} n_skipped={n_skipped}",
+                stack_trace=traceback.format_exc(),
+            ),
+        ),
+        # collecting the error
+        self.test_error_collector.collect(test_error)
+        # check for the patern presence for the collected errors
+        for error in self.test_error_collector.errors:
```

Comment:
Would you mind adding a case where the collector handles multiple errors?

Reply:
Added `invalid_csv_multi_scenario`.

Reply:
Would you mind adding an assertion about the length of `for error in self.test_error_collector.errors`? If it's empty we'll never hit this assertion.

Reply:
Done.